### PR TITLE
Priority filtering for syslog widget

### PR DESCRIPTION
**Pull request by Talkabout**
HI all,

I am currently in the process of migrating from Observium to LibreNMS. One of the main features I used in Observium was the syslog widget to show only critical logs. I was not able to achieve the same with the current LibreNMS widget so I started to dig a little bit into the implementation and finally found an approach to do exactly what I wanted. With this pull request the syslog widget receives an additional option allowing to define the priority level of log entries to be shown:

image

As this is my first change on the LibreNMS code I am pretty sure the code is still optimizable. Suggestions are very welcome!

Thanks!

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by Talkabout** [comment]
Hi @PipoCanaja, @murrant ,

second try with this pull request :) Hopefully now everything is correct.

Thanks for your help!

**Comment by Talkabout** [comment]
Hi @murrant ,

as you suggested in the previous pull request I have changed the "PriorityController" to return a static list instead of a dynamic query. Not sure if the code/approach is correct or there is a better way to do this.

Thanks!

**Comment by murrant** [comment]
@Talkabout thanks, although I guess I wasn't clear enough :)

You really just need a regular select without using select2 to fetch data from the backend since this is just a static list.
So you can completely delete PriorityController and related items and just list out the options in html

**Comment by Talkabout** [comment]
Hi @murrant ,

you are totally right, it makes perfect sense! I was not sure if the backend logic is required for all options so I never considered that. But now looking also at other implementations (widgets) I can see, that this is quite common. I updated the code accordingly.

Thanks!

File: resources/views/widgets/settings/syslog.blade.php [hunk 1]
```diff
@@ -25,6 +25,16 @@
         <label for="hidenavigation-{{ $id }}" class="control-label">@lang('Hide Navigation')</label>
         <input type="checkbox" class="form-control" name="hidenavigation" id="hidenavigation-{{ $id }}" value="{{ $hidenavigation }}" data-size="normal" @if($hidenavigation) checked @endif>
     </div>
+    <div class="form-group">
+        <label for="level-{{ $id }}" class="control-label">@lang('Priority')</label>
+        <select class="form-control" name="level" id="level-{{ $id }}">
+            @if($level)
```
Comment (murrant) [inline resources/views/widgets/settings/syslog.blade.php:31]: why the @if($level) ?

File: app/Http/Controllers/Widgets/SyslogController.php [hunk 2]
```diff
@@ -43,6 +44,8 @@ public function getSettingsView(Request $request)
 
         $data['device'] = Device::hasAccess($request->user())->find($data['device']);
 
+        $data['priorities'] = array_flip(app('translator')->get('syslog.severity'));
```
Comment (murrant) [inline app/Http/Controllers/Widgets/SyslogController.php:47]: You don't need to array_flip() here, just flip $name and $val in syslog.blade.php foreach

File: resources/views/widgets/settings/syslog.blade.php (hunk 1 above)
Comment (Talkabout) [reply resources/views/widgets/settings/syslog.blade.php]: was a leftover from the previous approach, removed.

File: app/Http/Controllers/Widgets/SyslogController.php (hunk 2 above)
Comment (Talkabout) [reply app/Http/Controllers/Widgets/SyslogController.php]: Done as requested.

**Review by murrant — APPROVED** [review]
Looks good!  Thanks for seeing it through.

**Pull request merged**